### DISCUSSION TITLE
Simulation with active scalar field

### DISCUSSION THREAD

**Discussion by ayush-9s** [xcompact3d/Incompact3d#163] [Q&A] [answered]
Hi,

I am trying to simulate variable viscosity incompressible flows using xcompact3d. I have modified the transeq.f90 file to include the viscosity variation into the momentum equation rhs. The viscosity variation is governed by a scalar field(concentration) which is being solved for as well. I am getting fringe like patterns in the solution scalar distribution, while the velocity and pressure distributions do not have any such patterns. I wanted to ask if anyone knows why these kinds of fringes might appear in the solution. 

I have attached the concentration(fig 1) and x-velocity(fig 2) distributions for reference. I was giving a steep concentration distribution (zero in the jet, and one everywhere else) for this case. Can this be a problem? 
![concentration_dist_slice](https://user-images.githubusercontent.com/85876808/225836345-daf986ff-32fc-4d64-91ea-1e3891672c07.png)
![x-velocity_slice](https://user-images.githubusercontent.com/85876808/225836384-abf4eb75-751b-4b27-9985-709731a58d42.png)

Following is the plot of all the variables at a particular axial location in the centerplane. 
![plot_over_line](https://user-images.githubusercontent.com/85876808/225848708-75d325fd-ab34-458d-b2a6-9e2014ae5c4c.png)

This is a circular jet flow case I am testing, and it was working for incompressible constant viscosity simulations without any problems.

Any help will be highly appreciated.

Thanks,
Ayush.

**Comment by mathrack** [comment]
Hello, if the concentration field is discontinuous, the finite difference schemes can produce oscillations. I would suggest one of the following solutions :

- Use a smooth field
- Change the derivative operator to properly compute the derivative at discontinuities

EDIT : try to input a discontinuous signal and look at the first and second derivative

**Comment by slaizet** [comment]
Hi,
To complement Cedric' comment, it looks like you are using a zero velocity co-flow, which might not be a good idea. The shape of the oscillations (aligned with the mesh) suggests that you might want to increase your spatial resolution. 
Best,
Sylvain

**Comment by ayush-9s** [comment, marked as answer]
Hello,

Thanks a lot for the responses. I have tried using a more smooth concentration distribution, and it is reducing the fringes appearing in the solution. I will try with a more resolved grid and small co-flow as well.   

Thanks again,
Ayush.
